R/data_interface.R

```diff
@@ -422,21 +419,19 @@ spark_save_table <- function(x, path, mode = NULL, options = list()) {
 }
 
 #' @export
-spark_write_table.tbl_spark <- function(x, path, mode = NULL, options = list()) {
+spark_write_table.tbl_spark <- function(x, name, mode = NULL, options = list()) {
   sqlResult <- spark_sqlresult_from_dplyr(x)
   sc <- spark_connection(x)
 
-  operation <- if (spark_version(sc) < "2.0.0") "save" else "saveAsTable"
-  spark_data_write_generic(sqlResult, spark_normalize_path(path), operation, mode, options)
+  spark_data_write_generic(sqlResult, name, "saveAsTable", mode, options)
 }
 
 #' @export
-spark_write_table.spark_jobj <- function(x, path, mode = NULL, options = list()) {
+spark_write_table.spark_jobj <- function(x, name, mode = NULL, options = list()) {
   spark_expect_jobj_class(x, "org.apache.spark.sql.DataFrame")
   sc <- spark_connection(x)
 
-  operation <- if (spark_version(sc) < "2.0.0") "save" else "saveAsTable"
-  spark_data_write_generic(x, spark_normalize_path(path), operation, mode, options)
+  spark_data_write_generic(x, name, "saveAsTable", mode, options)
```

Comment:
This one actually works fine since by default everyone gets `r` permissions in the `hive` folder.